### ISSUE TITLE
Save disabled theme override properties for easy re-enabling

### ISSUE THREAD

**Issue by lostminds** [godotengine/godot-proposals#5765]
### Describe the project you are working on

Working with interface Controls nodes to design an interface

### Describe the problem or limitation you are having in your project

Theme override parameters on Control nodes (like a Label example below) have a checkbox to let you quickly toggle if the property in the theme should be overridden or not. The problem is that if you toggle this off, it also resets the override value. This is unexpected since the expected functionality of disable would be to temporarily disable the override, especially since there's also a reset button that has the same reset functionality now. 

This also means that it's harder to test toggling theme overrides on/off to compare the difference since toggling the override off resets the property and you need to enter the value/color/etc again (or use undo/redo as a workaround)

https://user-images.githubusercontent.com/17763524/201374432-2452237f-673b-46b2-98cb-bdb07e18abfe.mov

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Have override properties like these save their values even if you toggle the override toggle off, so that if you toggle them on again you instantly have the value back.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Either: Separate the enable bool state property from the value and don't change the actual stored value when enabled state changes, then set up a getter that returns empty if enabled is false so it acts as if it's empty if it's disabled.

Or, if you don't want to add a separate enabled property: Have the parameters store a disabled_value when the overreide parameter is disabled, used only in the editor to show the value in the control and to restore this value if it's enabled again.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I don't think so

### Is there a reason why this should be core and not an add-on in the asset library?

It seems to me like a basic component of the Inspector property editing UI.

**Comment by Calinou** [comment]
This requires storing the "enabled" state of each theme item separately from the property itself, so it's not trivial to implement. Right now, when you uncheck a checkbox, it reverts to the default value and ensures the override isn't persisted to the scene file.

**Comment by lostminds** [comment]
I see, I suspected as much when I saw this unexpected behavior. I guess the checkbox still provides some usability to indicate it's an override-type-parameter to the user. But having the separate checkbox sort of implies to the user the parameter has a separate enabled state. As it could also be useful if it also actually had, I think it would be better all things being equal. 

But I understand if this might not be a priority since it's more of a usability detail and there are probably other feature more important to work on.
